scripts/west_commands/blobs.py

```diff
@@ -0,0 +1,166 @@
+
+class Blobs(WestCommand):
+
+    def __init__(self):
+        super().__init__(
+            'blobs',
+            # Keep this in sync with the string in west-commands.yml.
+            'work with binary blobs',
+            'Work with binary blobs',
+            accepts_unknown_args=False)
+
+    def do_add_parser(self, parser_adder):
+        default_fmt = '{module}: {status} {path} ({type}, {abspath})'
+        parser = parser_adder.add_parser(
+            self.name,
+            help=self.help,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description=self.description,
+            epilog=textwrap.dedent(f'''\
+            FORMAT STRINGS
+            --------------
+
+            Blobs are listed using a Python 3 format string. Arguments
+            to the format string are accessed by name.
+
+            The default format string is:
+
+            "{default_fmt}"
+
+            The following arguments are available:
+
+            - module: name of the module that contains this blob
+            - abspath: blob absolute path
+            - status: short status (A: present, M: hash failure, D: not present)
+            - path: blob local path from <module>/zephyr/blobs/
+            - sha256: blob SHA256 hash in hex
+            - type: type of blob
+            - version: version string
+            - license_path: path to the license file for the blob
+            - uri: URI to the remote location of the blob
+            - description: blob text description
+            - doc-url: URL to the documentation for this blob
+            '''))
+
+        # Remember to update west-completion.bash if you add or remove
+        # flags
+        parser.add_argument('subcmd', nargs=1, choices=['list', 'fetch'],
+                            help='''Select the sub-command to execute.
+                            Currently only list and fetch are supported.''')
+
+        # Remember to update west-completion.bash if you add or remove
+        # flags
+        parser.add_argument('-f', '--format', default=default_fmt,
+                            help='''Format string to use to list each blob;
+                                    see FORMAT STRINGS below.''')
+
+        parser.add_argument('-m', '--modules', type=str, default="",
+                            help='''a comma-separated list of modules; only blobs whose
+                            names are on this list will be taken into account
+                            by the sub-command.''')
+        parser.add_argument('-a', '--all', action='store_true',
+                            help='use all modules.')
+ 
+        return parser
+
+    def get_status(self, path, sha256):
+        if not path.is_file():
+            return 'D'
+        with path.open('rb') as f: 
+            m = hashlib.sha256()
+            m.update(f.read())
+            if sha256.lower() == m.hexdigest():
+                return 'A'
+            else:
+                return 'M'
```

Comment:
Huh. I'm surprised so much boilerplate is required, but I can't find a simpler way in the hashlib documentation.